app/controllers/carts_controller.rb

```diff
@@ -1,15 +1,20 @@
 class CartsController < ApplicationController
+  before_action :logged_in_user, only: :add_to_cart
 
   def index
     @cart = get_cart
   end
 
   def add_to_cart
-    if logged_in?
-      add_product_to_cart_cookie(params[:product_id])
-      redirect_to root_path
-    else
-      redirect_to login_path
-    end
+    add_product_to_cart_cookie(params[:product_id])
```

Comment:
Why not just `add_product_to_cart`? :-)

Reply:
I think I wanted to make it clearer how it adds it, but it sounds better simpler. Thanks!

Reply:
The internals of a method shouldn't be exposed, just the behaviour - the internals (how it works) will change over time :-)

Reply:
True, re-reading that method name now, I know what you mean.